src/coprocessor/dag/rpn_expr/types/test_util.rs

```diff
@@ -36,6 +36,17 @@ impl RpnFnScalarEvaluator {
         self
     }
 
+    pub fn push_params<T, S>(mut self, values: S) -> Self
+    where
+        T: Into<ScalarValue>,
+        S: std::iter::IntoIterator<Item = T>,
```

Comment:
`IntoIterator` is in Rust's prelude.

And, what about `values: impl IntoIterator<Item = impl Into<ScalarValue>>`?

Reply:
Wow. I don't know this syntax is valid. I will give a try.

Reply:
Interesting..For some reasons I don't know, this leads to:

```
error[E0282]: type annotations needed
   --> src/coprocessor/dag/rpn_expr/impl_compare.rs:659:18
    |
659 |                 .push_params(args)
    |                  ^^^^^^^^^^^ cannot infer type for `T`

error: aborting due to previous error
```

Reply:
No problem here. I'm curious about why there is type `T` when the function looks like ` pub fn push_params(mut self, values: impl IntoIterator<Item = impl Into<ScalarValue>>) -> Self `